api/maven-api-core/src/main/java/org/apache/maven/api/Project.java

```diff
@@ -37,24 +37,103 @@
 @Experimental
 public interface Project {
 
+    /**
+     * Returns the project groupId.
+     */
     @Nonnull
     String getGroupId();
 
+    /**
+     * Returns the project artifactId.
+     */
     @Nonnull
     String getArtifactId();
 
+    /**
+     * Returns the project version.
+     */
     @Nonnull
     String getVersion();
 
+    /**
+     * Returns the project packaging.
+     * <p>
+     * Note: unlike in legacy code, logical checks against string representing packaging (returned by this method)
+     * are NOT recommended (code like {@code "pom".equals(project.getPackaging)} must be avoided). Use method
+     * {@link #getArtifacts()} to gain access to POM or build artifact.
+     *
+     * @see #getArtifacts()
+     */
     @Nonnull
     String getPackaging();
 
+    /**
+     * Returns the project POM artifact, that is the artifact of the POM of this project. Every project have POM
+     * artifact, while the existence of backing POM file is NOT a requirement (i.e. for some transient projects).
+     *
+     * @see org.apache.maven.api.services.ArtifactManager#getPath(Artifact)
+     */
+    @Nonnull
+    default Artifact getPomArtifact() {
+        return getArtifacts().get(0);
+    }
+
+    /**
+     * Returns the project artifact, that is the artifact produced by this project build. This artifact MAY be same
+     * as the one returned by {@link #getPomArtifact()}, if the project is actually not producing any main artifact.
+     * The existence of artifact backing file depends on which lifecycle step this method was invoked, as the file
+     * may not yet be built.
+     * <p>
+     * If only non-POM artifacts are needed, better use {@link #getArtifacts()} method: if that method returns list
+     * having one element, the methods {@link #getPomArtifact()} and this one will return same artifact. For non-POM
+     * artifacts one would filter for results having two elements, and would consume second element of the list.
```

Comment:
Should we add something like that to be complete and instead of that whole paragraph which is a bit complicated ?
```
@Nonnull
default Optional<Artifact> getMainArtifact() {
    List<Artifact> artifacts = getArtifacts();
    return artifacts.size() == 2 ? Optional.of(artifacts.get(1)) : Optional.empty();
}
```

Reply:
We may, but personally I dislike that "main" artifact is absent. This is the case when you have two variables *a* and *b* and special case of *a = b*.

Maybe the method name... should be BuildArtifact? ProducedArtifact?

Reply:
Changed my mind, did that.